docs/src/modules/components/AppFrame.js

```diff
@@ -28,23 +28,42 @@ import { pathnameToLanguage } from 'docs/src/modules/utils/helpers';
 import PageContext from 'docs/src/modules/components/PageContext';
 import { useUserLanguage, useTranslate } from 'docs/src/modules/utils/i18n';
 import LanguageIcon from '@mui/icons-material/Translate';
+import { debounce } from '@mui/material/utils';
 
 const LOCALES = { zh: 'zh-CN', pt: 'pt-BR', es: 'es-ES' };
 const CROWDIN_ROOT_URL = 'https://translate.mui.com/project/material-ui-docs/';
 
+const nProgressStart = debounce(() => {
+  NProgress.start();
+}, 300);
```

Comment:
@danilo-leal What do you think about 300ms for a UX/UI perspective? I proposed this value in #22486, based on theoretical values: https://mui.com/components/progress/#delaying-appearance. But I didn't try the experience.

Reply:
I have tried the UX, 300ms works for me. However, it seems that this 300ms only accounts for the time it takes to load the page over the network, it doesn't account for the JavaScript-blocking rendering time. 

1. Open https://deploy-preview-30087--material-ui.netlify.app/components/icons/
2 Click on "Material Icons"
3. See how it takes 800ms for the page to transition:

<img width="539" alt="Screenshot 2021-12-07 at 13 19 00" src="https://user-images.githubusercontent.com/3165635/145027823-43dd8601-ec8c-4e3e-89e7-2deb73c8c4d1.png">

4. No progress bar

I'm not sure we can do much on our side.

Reply:
I came across this snippet where they add a timeout o 250ms before stopping the animation: https://gist.github.com/jaydenseric/15a61ecfe3b52599409787c33fcfe9da#file-routeindicator-js-L21

Reply:
As a general rule of thumb, for "productive" motion (see [Duration, 'Time Scales'](https://www.designsystems.com/5-steps-for-including-motion-design-in-your-system/)), I'd go with a little less, 200ms. But I couldn't try myself either, so that's just a general guideline.

Reply:
I don't think that it's about motion nor animation here, but about https://www.nngroup.com/articles/response-times-3-important-limits/.

Reply:
The general ID of hiding the progress bar for very quick page switch seems good to me :+1:

Reply:
Have updated it to 200ms